internal/controllers/configuration/secret_controller.go

```diff
@@ -113,6 +113,13 @@ func (r *CoreV1SecretReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 		return ctrl.Result{}, err
 	}
+
+	err := util.PopulateTypeMeta(secret)
```

Comment:
I was wondering whether we could make it somehow so that we do not have to remember to call this in every new controller. I prepared a suggestion that decorates the client we pass to every controller so it's more DRY: https://github.com/Kong/kubernetes-ingress-controller/pull/4832. It also uses the client's schema instead of building it on every call. WDYT?

Reply:
Yeah, that makes sense. Was there anything further you were planning to add to it? It's still marked as a draft.

Reply:
I was hopeful too soon. Building off https://github.com/Kong/kubernetes-ingress-controller/pull/4832#issuecomment-1760282746, I've merged that with some additional changes:

- Instead of providing a wrapped client to the controllers, apply it once to the manager. Docs say this is spooky voodoo, but I think it should be fine given that 80% of it is still the normal client. [I asked controller-runtime](https://kubernetes.slack.com/archives/C02MRBMN00Z/p1697144118238979) if they know of anything more we'd need to do this properly.
- Restored the scheme get and TypeMeta call to the key func. It [doesn't use the manager client](https://github.com/Kong/kubernetes-ingress-controller/pull/4832#issuecomment-1760431813) and IDK if we can stuff a workaround in the client-go cache. I wanted to try and get the scheme only once in CacheIndexers initialization, but doing so looked incredibly cumbersome because a bunch of standalone non-method functions need to call `objectKeyFunc()` also.

Reply:
Sounds good to me - looking into controller-runtime code the default for `opts.NewClient` is just `client.New` so we're functionally equal with that now.